File: app/src/main/java/com/datarockets/mnchkn/activities/onboard/di/OnboardComponent.java
package com.datarockets.mnchkn.activities.onboard.di;

import com.datarockets.mnchkn.activities.onboard.OnboardActivity;
import com.datarockets.mnchkn.activities.onboard.OnboardPresenter;
import com.datarockets.mnchkn.di.ActivityScope;
import com.datarockets.mnchkn.di.AppComponent;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = OnboardModule.class)
public interface OnboardComponent {
    void inject(OnboardActivity onboardActivity);
    OnboardPresenter getOnboardPresenter();
}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/onboard/di/OnboardComponent.kt
package com.datarockets.mnchkn.activities.onboard.di

import com.datarockets.mnchkn.activities.onboard.OnboardActivity
import com.datarockets.mnchkn.activities.onboard.OnboardPresenter
import com.datarockets.mnchkn.di.ActivityScope
import com.datarockets.mnchkn.di.AppComponent

import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(OnboardModule::class))
interface OnboardComponent {
    fun inject(onboardActivity: OnboardActivity)
    val onboardPresenter: OnboardPresenter
}
